internal/pkg/controllers/selinuxpolicy/configmap_controller.go

```diff
@@ -0,0 +1,290 @@
+		if err := r.client.Status().Update(context.TODO(), policyCopy); err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "Updating policy without status")
+		}
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	reqLogger.Info("Reconciling pods for policy")
+
+	nodesList := &corev1.NodeList{}
+	if err := r.client.List(context.TODO(), nodesList); err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "Getting node list")
+	}
+	for nodeidx := range nodesList.Items {
+		// Define a new Pod object
+		pod := newPodForPolicy(policyName, policyNamespace, &nodesList.Items[nodeidx])
+		if err := controllerutil.SetControllerReference(cminstance, pod, r.scheme); err != nil {
+			log.Error(err, "Failed to set pod ownership", "pod", pod)
+			return reconcile.Result{}, errors.Wrap(err, "Setting controller reference for SELinux installer pod")
+		}
+
+		// Check if this Pod already exists
+		found := &corev1.Pod{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, found)
+		if err != nil && kerrors.IsNotFound(err) {
+			reqLogger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+			if err = r.client.Create(context.TODO(), pod); err != nil {
+				return reconcile.Result{}, IgnoreAlreadyExists(err)
+			}
+			continue
+		} else if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "Gettting selinux installer Pod")
+		}
+
+		exitCode, gotCode := r.getInstallerContainerExitCode(found)
+		// Pod is still running - requeue
+		if !gotCode {
+			reqLogger.Info("Pod still running", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+			return reconcile.Result{Requeue: true, RequeueAfter: defaultRequeueTime}, nil
+		}
+		if exitCode != 0 {
+			reqLogger.Info("Pod failed", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name, "exit-code", exitCode)
+			policyCopy.Status.State = spov1alpha1.PolicyStateError
+			if err := r.client.Status().Update(context.TODO(), policyCopy); err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "Updating SELinux policy with install error")
+			}
+			return reconcile.Result{}, nil
+		}
+	}
+	policyCopy.Status.State = spov1alpha1.PolicyStateInstalled
+	if err := r.client.Status().Update(context.TODO(), policyCopy); err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "Updating SELinux policy with installation success")
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileConfigMap) getInstallerContainerExitCode(pod *corev1.Pod) (int32, bool) {
+	for idx := range pod.Status.ContainerStatuses {
+		containerStatus := &pod.Status.ContainerStatuses[idx]
+		if containerStatus.Name == "policy-installer" {
+			if containerStatus.State.Terminated != nil {
+				termState := containerStatus.State.Terminated
+				return termState.ExitCode, true
+			}
+		}
+	}
+	return -1, false
+}
+
+// newPodForPolicy returns a busybox pod with the same name/namespace as the cr.
```

Comment:
Is it a busybox pod? It looks like it's using the udica image for both containers

Reply:
I actually don't remember isf it's busybox or UBI haha. We'll actually change this implementation of the operator to spawn a single workload as opposed to a workload per policy. I'll start updating that when we have a decent proposal. First I'd like to start adding unit tests though